packages/zoe/tools/priceAuthorityRegistry.js

```diff
@@ -96,76 +105,118 @@ export const makePriceAuthorityRegistry = () => {
    *
    * @type {PriceAuthority}
    */
-  const priceAuthority = Far('fake price authority', {
-    async getQuoteIssuer(brandIn, brandOut) {
-      return E(paFor(brandIn, brandOut)).getQuoteIssuer(brandIn, brandOut);
-    },
-    async getTimerService(brandIn, brandOut) {
-      return E(paFor(brandIn, brandOut)).getTimerService(brandIn, brandOut);
-    },
-    async quoteGiven(amountIn, brandOut) {
-      return E(paFor(amountIn.brand, brandOut)).quoteGiven(amountIn, brandOut);
+  const priceAuthority = prepareExo(
+    baggage,
+    'composite price authority',
+    M.interface('compositePriceAuthority', {
+      getQuoteIssuer: M.call(BrandShape, BrandShape).returns(M.promise()),
+      getTimerService: M.call(BrandShape, BrandShape).returns(M.promise()),
+      quoteGiven: M.call(AmountShape, BrandShape).returns(M.promise()),
+      quoteWanted: M.call(BrandShape, AmountShape).returns(M.promise()),
+      makeQuoteNotifier: M.call(AmountShape, BrandShape).returns(M.promise()),
+      quoteAtTime: M.call(TimestampShape, AmountShape, BrandShape).returns(
+        M.promise(),
+      ),
+      quoteWhenLT: GuardCallAmountTuple,
+      quoteWhenLTE: GuardCallAmountTuple,
+      quoteWhenGTE: GuardCallAmountTuple,
+      quoteWhenGT: GuardCallAmountTuple,
+      mutableQuoteWhenLT: GuardCallAmountTuple,
+      mutableQuoteWhenLTE: GuardCallAmountTuple,
+      mutableQuoteWhenGTE: GuardCallAmountTuple,
+      mutableQuoteWhenGT: GuardCallAmountTuple,
+    }),
```

Comment:
I'm surprised we haven't written this interface guard before. The fluxAggregator is an exo price authority, isn't it?

Hm... there's one price authority held in the `prepareFluxAggregatorKit` closure...

https://github.com/Agoric/agoric-sdk/blob/6e658d97b098716210019f147fb764edc4337bc9/packages/inter-protocol/src/price/fluxAggregatorKit.js#L178-L179

It seems to be referenced by all instances of the `fluxAggregator` exo class kit.

https://github.com/Agoric/agoric-sdk/blob/6e658d97b098716210019f147fb764edc4337bc9/packages/inter-protocol/src/price/fluxAggregatorKit.js#L204-L206

Maybe that's OK, since there's only one instance of that exo class... but then shouldn't it be an exo singleton?

Reply:
It seems like [scaledPriceAuthority.js](https://github.com/Agoric/agoric-sdk/blob/master/packages/zoe/src/contracts/scaledPriceAuthority.js) should also use this interface guard. Is making that durable in our plans anywhere?

Reply:
for scaledPriceAuthority I added some docs: [34d2d7d](https://github.com/Agoric/agoric-sdk/pull/7555/commits/34d2d7d0ecf180744e29ef52a7e6073bd535058b)

for the Exo singleton idea let's chat